### ISSUE TITLE
Protobuf types missing in java_export sources/javadocs

### ISSUE THREAD

**Issue by jfancher** [bazelbuild/rules_jvm_external#462]
Seeing #448 reminded me to file this, although it's a slightly different issue.

This affects `examples/java-export` in this repo -- it defines a protobuf library, but those types only appear in the class jar, not the sources or javadocs.

The reason for this is bazelbuild/bazel#11830. However, that was flagged as "not a bug" for some reason; I think I disagree, but maybe it's possible to fix on this side. I was able to make it work with this:

```diff
--- a/private/rules/has_maven_deps.bzl
+++ b/private/rules/has_maven_deps.bzl
@@ -92,7 +92,11 @@ def _has_maven_deps_impl(target, ctx):
     items = _set_diff(all_jars.to_list(), jars_from_maven_deps.to_list())
     artifact_jars = depset(_filter_external_jars(ctx.workspace_name, items))

-    all_source_jars = target[JavaInfo].transitive_source_jars
+    all_source_jars = depset(
+        direct = target[JavaInfo].source_jars,
+        transitive = [target[JavaInfo].transitive_source_jars] + [i.artifact_source_jars for i in all_infos],
+    )
```

That seems like a hack, but maybe it's okay, or there's a nicer version of this fix.

**Comment by shs96c** [comment]
Including all the transitive source jars would lead to unexpectedly heavy source jars being generated, so it may be best to avoid that.

The protobuf libraries are generated using aspects, and I suspect that's part of the problem. I'm working on a change to how we calculate the contents of the maven jars which calculates the difference in the sets of transitive inputs. If that works, I think we can apply the same logic for source jars too.

**Comment by jfancher** [comment]
Oh, I haven't closed this because I haven't had a chance to confirm it, but the Bazel issue I created did get picked up and fixed. So I think this will work without further changes when that makes it into a release.

Thanks for looking into it though!

**Comment by jfancher** [comment]
Oh, I spoke too soon... I just noticed that the fix was subsequently rolled back.

**Comment by shs96c** [comment]
I have #475 open for this issue. I _think_ that fixes this.
